src/rust/include/rust/orchard/keys.h

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2021 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef ZCASH_RUST_INCLUDE_RUST_ORCHARD_KEYS_H
+#define ZCASH_RUST_INCLUDE_RUST_ORCHARD_KEYS_H
+
+#include "rust/streams.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// A pointer to an Orchard shielded payment address, as defined in
+// https://zips.z.cash/protocol/nu5.pdf#orchardpaymentaddrencoding
+struct OrchardRawAddressPtr;
+typedef struct OrchardRawAddressPtr OrchardRawAddressPtr;
+
+// Clones the given Orchard payment address and returns
+// a pointer to the newly created value. Both the original
+// one's memory and the newly allocated one need to be freed
+// independently.
+OrchardRawAddressPtr* orchard_address_clone(
+        const OrchardRawAddressPtr* ptr);
+
+// Free the memory allocated for the given Orchard payment address
+void orchard_address_free(OrchardRawAddressPtr* ptr);
+
+/// Parses an Orchard payment address from the given stream.
+///
+/// - If no error occurs, `payment_address_ret` will point to a Rust-allocated Orchard payment address.
+/// - If an error occurs, `payment_address_ret` will be unaltered.
+bool orchard_address_parse(
+    void* stream,
+    read_callback_t read_cb,
+    OrchardRawAddressPtr** payment_address_ret);
```

Comment:
This doesn't match the implementation. Fix would be:
```suggestion
/// Returns null if an error occurs.
OrchardRawAddressPtr* orchard_address_parse(
    void* stream,
    read_callback_t read_cb);
```